cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientFactory.java

```diff
@@ -77,54 +75,52 @@ public CloudFoundryClientFactory getCloudFoundryClientFactory() {
         return builder.build();
     }
 
-    public RestTemplate getGeneralPurposeRestTemplate() {
-        return restUtil.createRestTemplate(getHttpProxyConfiguration(), shouldTrustSelfSignedCertificates());
-    }
-
-    public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials) {
-        return createClient(controllerUrl, credentials, (CloudSpace) null);
-    }
-
-    public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, OAuthClient oAuthClient) {
-        return createClient(controllerUrl, credentials, null, oAuthClient);
+    public WebClient getGeneralPurposeWebClient() {
+        return restUtil.createWebClient(shouldTrustSelfSignedCertificates());
     }
 
     public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, String organizationName,
-                                                  String spaceName) {
-        return createClient(controllerUrl, credentials, organizationName, spaceName,
-                            createOAuthClient(controllerUrl, credentials.getOrigin()));
-    }
-
-    public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, String organizationName,
-                                                  String spaceName, OAuthClient oAuthClient) {
+                                                  String spaceName, OAuthClient oAuthClient, List<ExchangeFilterFunction> exchangeFilters) {
         CloudControllerRestClient clientWithoutTarget = createClient(controllerUrl, credentials, oAuthClient);
         CloudSpace target = clientWithoutTarget.getSpace(organizationName, spaceName);
 
-        return createClient(controllerUrl, credentials, target, oAuthClient);
+        return createClient(controllerUrl, credentials, target, oAuthClient, exchangeFilters);
     }
 
-    public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, CloudSpace target) {
-        return createClient(controllerUrl, credentials, target, createOAuthClient(controllerUrl, credentials.getOrigin()));
+    public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, OAuthClient oAuthClient) {
+        return createClient(controllerUrl, credentials, null, oAuthClient, null);
```

Comment:
Why not pass `Collections.emptyList()` as a value for `exchangeFilters`? It'll save you a `null` check later on.